plugins/inputs/http_listener/http_listener.go

```diff
@@ -158,8 +161,134 @@ func (t *HttpListener) ServeHTTP(res http.ResponseWriter, req *http.Request) {
 	}
 }
 
+func (h *HttpListener) serveWrite(res http.ResponseWriter, req *http.Request) {
+	// Check that the content length is not too large for us to handle.
+	if req.ContentLength > h.MaxBodySize {
+		tooLarge(res)
+		return
+	}
+	now := time.Now()
+
+	// Handle gzip request bodies
+	body := req.Body
+	if req.Header.Get("Content-Encoding") == "gzip" {
+		body, err := gzip.NewReader(req.Body)
+		defer body.Close()
+		if err != nil {
+			log.Println("E! " + err.Error())
+			badRequest(res)
+			return
+		}
+	}
+	body = http.MaxBytesReader(res, body, h.MaxBodySize)
+
+	var return400 bool
+	var hangingBytes bool
+	buf := h.pool.get()
+	defer func() { h.pool.put(buf) }()
```

Comment:
`defer h.pool.put(buf)` will work, unless you want to delay evaluation of the captured variables, which is probably unnecessary